File: Minimum_Equal_Sum_of_Two_Arrays_After_Replacing_Zeros.java
class Solution {
    public long minSum(List<Integer> A, List<Integer> B) {
        long asum = 0, bsum = 0, a0 = 0, b0 = 0;
        for (int a : A) {
            a0 += a == 0 ? 1 : 0;
            asum += Math.max(a, 1);
        }
        for (int b : B) {
            b0 += b == 0 ? 1 : 0;
            bsum += Math.max(b, 1);
        }
        if (asum < bsum && a0 == 0)
            return -1;
        if (asum > bsum && b0 == 0)
            return -1;
        return Math.max(asum, bsum);
    }
}

File: Minimum_Equal_Sum_of_Two_Arrays_After_Replacing_Zeros.py
class Solution:
    def minSum(self, A, B):
        asum = 0
        bsum = 0
        a0 = 0
        b0 = 0
        for a in A:
            a0 += a == 0
            asum += max(a, 1)
        for b in B:
            b0 += b == 0
            bsum += max(b, 1)
        if asum < bsum and a0 == 0:
            return -1
        if asum > bsum and b0 == 0:
            return -1
        return max(asum, bsum)
